source/frontend/src/components/maps/leaflet/LayerPopup/LayerPopupContainer.tsx

```diff
@@ -50,160 +49,189 @@ const emptyLayerPopupInformation: LayerPopupInformation = {
   layers: [],
 };
 
-export const LayerPopupContainer = React.forwardRef<LeafletPopup, React.PropsWithChildren<unknown>>(
-  (props, ref) => {
-    const mapMachine = useMapStateMachine();
+interface ILayerPopupContainer {
+  featureDataset: SinglePropertyFeatureDataSet | null;
+}
 
-    const [layerPopup, setLayerPopup] = useState<LayerPopupInformation>(emptyLayerPopupInformation);
+export const LayerPopupContainer: React.FC<
+  React.PropsWithChildren<ILayerPopupContainer>
+> = props => {
+  const featureSet = props.featureDataset;
 
-    useEffect(() => {
-      if (mapMachine.mapLocationFeatureDataset) {
-        const featureSet = mapMachine.mapLocationFeatureDataset;
+  const [layerPopup, setLayerPopup] = useState<LayerPopupInformation>(emptyLayerPopupInformation);
 
-        const layersData: LayerData[] = [];
+  useEffect(() => {
+    if (featureSet) {
+      const layersData: LayerData[] = [];
 
-        if (exists(featureSet.parcelFeature)) {
-          const parcelData: LayerData = { title: 'LTSA ParcelMap data', data: null, config: {} };
+      if (exists(featureSet.parcelFeature)) {
+        const parcelData: LayerData = { title: 'LTSA ParcelMap data', data: null, config: {} };
 
-          parcelData.bounds = featureSet.parcelFeature.geometry
-            ? geoJSON(featureSet.parcelFeature.geometry).getBounds()
-            : undefined;
-          parcelData.config = parcelLayerPopupConfig;
-          parcelData.data = featureSet.parcelFeature.properties;
-          parcelData.feature = featureSet.parcelFeature;
-          layersData.push(parcelData);
-        }
-        if (exists(featureSet.crownLandLeasesFeature)) {
+        parcelData.bounds = featureSet.parcelFeature.geometry
+          ? geoJSON(featureSet.parcelFeature.geometry).getBounds()
+          : undefined;
+        parcelData.config = parcelLayerPopupConfig;
+        parcelData.data = featureSet.parcelFeature.properties;
+        parcelData.feature = featureSet.parcelFeature;
+        layersData.push(parcelData);
+      }
+
+      if (
+        exists(featureSet.crownLandLeasesFeatures) &&
+        featureSet.crownLandLeasesFeatures.length > 0
+      ) {
+        featureSet.crownLandLeasesFeatures.forEach((crownLandLeasesFeature, index) => {
           const parcelData: LayerData = {
-            title: 'Crown Land Leases',
+            title: `Crown Land Leases (${index + 1} of ${featureSet.municipalityFeatures.length})`,
```

Comment:
this seems like the wrong feature set: featureSet.municipalityFeatures.length

Reply:
good catch